exist-core/src/main/java/org/exist/test/runner/AbstractTestRunner.java

```diff
@@ -111,4 +133,50 @@ protected static Sequence executeQuery(final Source query, final List<Function<X
             }
         }
     }
+
+    protected final BrokerPool getBrokerPool() {
+        return requireNonNull(existServer.getBrokerPool(), "brokerPool is null");
+    }
+
+    protected static String checkDescription(Object source,  String description) {
+        if (description == null) {
+            throw new IllegalArgumentException(source + " description is null");
+        }
+        if (description.isEmpty()) {
+            throw new IllegalArgumentException(source + " description is empty");
+        }
+        if (description.startsWith("(")) {
+            throw new IllegalArgumentException(source + " description '" + description + "' starts with '('");
+        }
+        return description;
+    }
+
+    private static class ExistServer {
+        private ExistEmbeddedServer embeddedServer;
+
+        protected void stopServer() {
+            if (embeddedServer != null && embeddedServer.getBrokerPool() != null) {
+                embeddedServer.stopDb();
+                embeddedServer = null;
+            }
+        }
+
+        protected BrokerPool getBrokerPool() {
+            if (embeddedServer == null) {
+                embeddedServer = new ExistEmbeddedServer(true, true);
+                try {
+                    embeddedServer.startDb();
+                } catch (DatabaseConfigurationException | EXistException | IOException e) {
+                    throw new AssertionError("Embedded eXist server failed to start ", e);
+                }
+            } else {
+                try {
+                    embeddedServer.restart();
```

Comment:
This seems like every call to getBrokerPool() will restart the database. This seems like it would be very expensive.

Is it that this method is not really named correctly, or that we don't want to restart the database on every getBrokerPool call?

I need to understand this a bit better...

Reply:
I can do a try leaving this away...

Reply:
Seem that when letting this away it leads to non available security manager service though..